lib/models/Player.js

```diff
@@ -598,10 +598,40 @@ Player.prototype.browse = function browse(objectId, startIndex, limit) {
     });
 };
 
+Player.prototype.browseAll = function browseAll(objectId) {
+  let result = {
+    items: [],
+    startIndex: 0,
+    numberReturned: 0,
+    totalMatches: 1
+  };
+
+  let getChunk = (chunk) => {
+    Array.prototype.push.apply(result.items, chunk.items);
+    result.numberReturned += chunk.numberReturned;
+    result.totalMatches = chunk.totalMatches;
+
+    if (chunk.startIndex + chunk.numberReturned >= chunk.totalMatches) {
+      return result;
+    }
+
+    // Recursive promise chain
+    return this.browse(objectId, chunk.startIndex + chunk.numberReturned, 0)
+        .then(getChunk);
+  };
+
+  return Promise.resolve(result)
+      .then(getChunk);
+};
+
 Player.prototype.getQueue = function getQueue(startIndex, limit) {
   return this.browse('Q:0', startIndex, limit);
 };
 
+Player.prototype.getCompleteQueue = function getCompleteQueue() {
+  return this.browseAll('Q:0').then(result => result.items);
+};
+
 Player.prototype.toJSON = function toJSON() {
```

Comment:
It would be fine that getQueue would return everything. That's how it is used in the http api at least.